go/adbc/driver/databricks/schema.go

```diff
@@ -128,182 +194,287 @@ func ResultSchemaToArrowSchema(dbxSchema *sql.ResultSchema) (*arrow.Schema, erro
+	if !strings.HasPrefix(s, prefix+"<") || !strings.HasSuffix(s, ">") {
+		return "", false
 	}
-	switch col.TypeName {
-	case sql.ColumnInfoTypeNameDecimal:
-		precision, scale := col.TypePrecision, col.TypeScale
-		if precision == 0 {
-			// Mostly used by the recursive case (array, map, etc)
-			// "DECIMAL(p,s) and DECIMAL(p) with scale default at 0"
-			if matches := regexp.MustCompile(`DECIMAL\((\d+)(?:,?(\d+))?\)`).FindStringSubmatch(col.TypeText); matches != nil {
-				var err error
-				var precision64 int64
-				precision64, err = strconv.ParseInt(matches[1], 10, 8)
-				if err != nil {
-					return nil, fmt.Errorf("invalid decimal precision: %v", err)
-				}
-				precision = int(precision64)
-
-				// expect 3 matches total: DECIMAL, p, s
-				if len(matches) == 3 {
-					var scale64 int64
-					scale64, err = strconv.ParseInt(matches[2], 10, 8)
-					if err != nil {
-						return nil, fmt.Errorf("invalid decimal scale: %v", err)
-					}
-					scale = int(scale64)
-				}
+	start := len(prefix) + 1
+	depth := 0
+	for i := start; i < len(s); i++ {
+		if s[i] == '<' {
+			depth++
+		} else if s[i] == '>' {
+			if depth == 0 {
+				// Found the matching closing bracket
+				return s[start:i], true
 			}
+			depth--
 		}
-		if precision <= 38 {
-			return &arrow.Decimal128Type{Precision: int32(precision), Scale: int32(scale)}, nil
-		}
-		return &arrow.Decimal256Type{Precision: int32(precision), Scale: int32(scale)}, nil
-	// todo(jasonlin45): Add support for TIMESTAMP_NTZ
-	case sql.ColumnInfoTypeNameTimestamp:
+	}
+	// If we get here, brackets were unbalanced
+	return "", false
+}
+
+func getArrowTypeFromStringType(col string) (arrow.DataType, error) {
+	// simple types - base case
+	if arrowType, ok := stringTypeToArrowTypeMap[col]; ok {
+		return arrowType, nil
+	}
+	if col == "TIMESTAMP" || col == "TIMESTAMP_NTZ" {
 		return &arrow.TimestampType{
 			Unit:     arrow.Microsecond,
-			TimeZone: "Etc/UTC", // todo(jasonlin45): Support session timezone
+			TimeZone: "Etc/UTC",
 		}, nil
-	case sql.ColumnInfoTypeNameArray:
-		// Only available from the full SQL type spec
-		// parse from "ARRAY<elementType>"
-		if matches := regexp.MustCompile(`ARRAY<(.+)>`).FindStringSubmatch(col.TypeText); matches != nil {
-			elementType := strings.TrimSpace(matches[1])
-			// Create a temporary ColumnInfo to recursively parse the element type
-			elementCol := sql.ColumnInfo{
-				TypeName: sql.ColumnInfoTypeName(elementType),
-			}
+	}
+	// { DECIMAL | DEC | NUMERIC } [ (  p [ , s ] ) ]
+	if matches := regexp.MustCompile(decimalTypeRegex).FindStringSubmatch(col); matches != nil {
+		var err error
+		var precision64 int64
+		precision64, err = strconv.ParseInt(matches[1], 10, 8)
+		if err != nil {
+			return nil, fmt.Errorf("invalid decimal precision: %v", err)
+		}
+		precision := int(precision64)
 
-			elementArrowType, err := getArrowTypeFromColumnInfo(elementCol)
+		var scale = 0
+		if len(matches) > 2 && matches[2] != "" {
```

Comment:
So this is saying it has a scale but you're passing the burden of this knowledge onto the reader which feels like code smell. Worst call add a comment this is explicitly for handling scale

Reply:
how would you do it then?

Reply:
```suggestion
                // Match scale if present
                if len(matches) > 2 && matches[2] != "" {
```

maybe this? Helps identify what these ops are doing